adapters/synacormedia/synacormedia.go

```diff
@@ -72,11 +75,12 @@ func (a *SynacorMediaAdapter) makeRequest(request *openrtb.BidRequest) (*adapter
 
 	var err error
 
-	if firstExtImp == nil || firstExtImp.SeatId == "" {
+	if firstExtImp == nil || firstExtImp.SeatId == "" || firstExtImp.TagId == "" {
 		return nil, append(errs, &errortypes.BadServerResponse{
-			Message: fmt.Sprintf("Impression missing seat id"),
+			Message: fmt.Sprintf("Invalid Impression"),
 		})
 	}
+	// this is where the empty seatId is filled
 	re = &ReqExt{SeatId: firstExtImp.SeatId}
```

Comment:
Isn't `TagId` also meant to be copied into the `ReqExt` object? I realize `ReqExt` isn't even defined with a `TagId` field, I'm just wondering if we don't want that value in there.

Reply:
TagId is a part of the Imp object, so for each Imp in request.Imp we get the TagId out of the Ext object and add it to the root of the Imp object and then add all those Imps to validImps.  Those become the new request.Imp that gets sent out.

Reply:
You may wish to consider using a different error message here and on line 67 so the unique error message may assist in debugging.